lib/dl_core/dl_core/fields.py

```diff
@@ -474,3 +474,10 @@ def remove_multiple(self, field_ids: Collection[str]) -> None:
         for field in fields_to_remove:
             self.fields.remove(field)
         self.clear_caches()
+
+    def get_fields_by_source(self, source_id: str) -> List[BIField]:
```

Comment:
get_***direct***_fields_by_source?

Reply:
looks like only direct_fields can be retrieved by source so get_direct_fields_by_source seems equal

Reply:
I may use getattr instead of checking what column type is it

Reply:
nvm, changed this